sql/sql_tablespace.cc

```diff
@@ -483,11 +483,14 @@ bool Sql_cmd_create_tablespace::execute(THD *thd) {
         thd->variables.default_table_encryption == DEFAULT_TABLE_ENC_ON ||
         global_system_variables.default_table_encryption ==
             DEFAULT_TABLE_ENC_ONLINE_TO_KEYRING;
-    // We set encrypt_type, which is later assigned to tablespace's DD
-    // encryption option to Y for online KEYRING encryption. This field is
-    // designed so the user could check if given table is encrypted or not. The
-    // details on how it is encrypted (KEYRING in this case) are left in SE.
-    encrypt_type = encrypt_tablespace ? "Y" : "N";
+    if (encrypt_tablespace) {
+      encrypt_type = (global_system_variables.default_table_encryption ==
+                      DEFAULT_TABLE_ENC_ONLINE_TO_KEYRING)
+                         ? "ONLINE_KEYRING"
```

Comment:
I really hope we now allow foreground KEYRING encrypted general tablespaces. Because this code, does it automatically based on global option.

Reply:
This assigns encrypt_type to ONLINE_TO_KEYRING, not KEYRING and encrypt_type means something else for tablespaces and file_per_table tables.

Reply:
Ok, Generally speaking, is it only a "scope" limitation to support KEYRING for general tablespaces? CREATE TABLESPACE ts1 ADD DATAFILE ts1.ibd ENCRYPTION=KEYRING?